app/unit-tests/src/org/commcare/android/tests/application/AutoUpdateTest.java

```diff
@@ -38,92 +26,6 @@
 @Config(application = CommCareTestApplication.class)
 @RunWith(CommCareTestRunner.class)
 public class AutoUpdateTest {
-    private final static String REF_BASE_DIR = "jr://resource/commcare-apps/update_tests/";
-    private final static String username = "test";
-    private final static String password = "123";
-
-    /**
-     * Check that logging out after an auto-update check failed once should
-     * trigger the auto-update to resume.
-     */
-    @Test
-    public void testAppAutoUpdateLogoutRetry() {
```

Comment:
Is there a reason we aren't replicating a similar test? 

I think in general we should consider it a poor outcome to replace one process that was tested with another that had no tests.

Reply:
this test checks whether a failed update resumes after the logout. Though since we are currently not cancelling the UpdateWorker on logout but seat change, I don't think this test serves any purpose. Plus earlier we were using our own code to schedule the app updates which made sense to test. Though now since WorkManager takes care of scheduling, it's not guaranteed for an update to resume right away after login.

Reply:
Is there a best practice for how people test WorkManager related behaviors that Google advises? 

I agree that this wasn't a great test, but it did result in *some* code coverage of execution, so I'm a bit gun shy about that shift

Reply:
@ctsims In terms of code coverage of our update code we do have [AppUpdateTest](https://github.com/dimagi/commcare-android/blob/master/app/unit-tests/src/org/commcare/android/tests/application/AppUpdateTest.java) that runs through most update scenarios. 

For testing WorkManager itself, Android has some [instrumentation tests APIs ](https://developer.android.com/topic/libraries/architecture/workmanager/how-to/testing). I am trying to currently figure out how to use them with Roblectric but facing an issue described [here](https://stackoverflow.com/questions/60222892/modifying-robolectrics-instrumentation-configuration-with-androidx-tests).